src/core_modules/capture-core/components/ListView/ColumnSelector/DragDropList/DragDropListItem.component.js

```diff
@@ -1,107 +1,69 @@
+    const [, drop] = useDrop({
+        accept: ItemTypes.LISTITEM,
+        hover(item: { id: string, index: number }) {
+            const dragIndex = item.index;
+            const hoverIndex = index;
 
-        // Don't replace items with themselves.
-        if (dragIndex === hoverIndex) {
-            return;
-        }
+            // Don't replace items with themselves.
+            if (dragIndex === hoverIndex) {
+                return;
+            }
 
-        // Time to actually perform the action.
-        props.moveListItem(dragIndex, hoverIndex);
+            // Time to actually perform the action
+            moveListItem(dragIndex, hoverIndex);
 
-        // Note: we're mutating the monitor item here!
-        // Generally it's better to avoid mutations,
-        // but it's good here for the sake of performance
-        // to avoid expensive index searches.
-        monitor.getItem().index = hoverIndex;
-    },
-};
+            // Note: we're mutating the item here!
+            // Generally it's better to avoid mutations,
+            // but it's good here for the sake of performance
+            // to avoid expensive index searches.
+            item.index = hoverIndex;
+        },
+    });
 
-class Index extends Component<Props> {
-    render() {
-        const { text, isDragging, connectDragSource, connectDropTarget } = this.props;
-        const opacity = isDragging ? 0 : 1;
+    const [{ isDragging }, drag] = useDrag({
+        type: ItemTypes.LISTITEM,
+        item: { id, index },
+        collect: monitor => ({
+            isDragging: monitor.isDragging(),
+        }),
+    });
 
-        // $FlowFixMe[incompatible-extend] automated comment
-        return connectDropTarget(connectDragSource(
-            <tr key={this.props.id} tabIndex={-1} style={{ ...style, opacity }}>
-                <TableCell component="th" scope="row">
-                    <Checkbox
-                        checked={this.props.visible}
-                        tabIndex={-1}
-                        onChange={this.props.handleToggle(this.props.id)}
-                        label={text}
-                        className={this.props.classes.checkbox}
-                        valid
-                        dense
-                    />
-                </TableCell>
-                <TableCell>
-                    <span style={{ float: 'right' }}>
-                        <IconReorder24 />
-                    </span>
-                </TableCell>
-            </tr>,
-        ));
-    }
-}
+    drag(drop(ref));
 
-export const DragDropListItemPlain =
-    DragSource(
-        ItemTypes.LISTITEM,
-        cardSource,
-        (connect, monitor) => ({ connectDragSource: connect.dragSource(), isDragging: monitor.isDragging() }),
-    )(DropTarget(
-        ItemTypes.LISTITEM,
-        cardTarget,
-        connect => ({ connectDropTarget: connect.dropTarget() }),
-    )(Index));
+    const opacity = isDragging ? 0.5 : 1;
```

Comment:
> Note: I can see that the hover effect can be confusing under drag, but I found it hard to fix it.

How about setting the opacity to `0` when dragging? This way the drawn row will look empty and can help with the confusion of the hover effect (this is also how it currently works in play with the material UI components).

Reply:
<img width="404" alt="Skjermbilde 2024-07-31 kl  11 35 33" src="https://github.com/user-attachments/assets/3f5d8b22-ac2d-4018-8a13-62d54400221d">
I agree that setting the opacity to `0` makes it better so I will change it to `const opacity = isDragging ? 0 : 1;`. However, the **table row hover** is still confusing. I added a picture where I am dragging the _Gender_ , but the hover is marking _Last name_.

Reply:
I use Firefox where the hover issue doesn't happen, but I was able to reproduce it in Chrome. I bealive you can remove the hover styles altogether with something like this in the `DragDropListItem.component.js`.

// the styles
```
const styles = {
    rowWithoutHover: {
        '&:hover > td': {
            backgroundColor: 'transparent !important',
        },
    },
};
```

// and in the render use the class
```
return (
        <DataTableRow ref={ref} style={{ opacity }} className={classes.rowWithoutHover} draggable>
            <DataTableCell>{text}</DataTableCell>
            ......
```

Reply:
Thank you, @simonadomnisoru! That works. With the changes, the hover was completely removed, so in addition to your suggested changes, I added a state to tell if any element in the list is being dragged. Based on that state, I will disable the hover. What do you think? I appreciate your help!

Reply:
It looks great now 👏